cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -839,14 +854,16 @@ public void workflowSchedules(HttpRequest request, HttpResponder responder,
    * Get schedule state.
    */
   @GET
-  @Path("/apps/{app-id}/workflows/{workflow-id}/schedules/{schedule-id}/status")
+  @Path("/apps/{app-id}/workflows/{workflow-id}/schedules/{schedule-name}/status")
```

Comment:
The endpoint "apps/{app-id}/schedules/{schedule-name}" clashes with the endpoint "/apps/{app-id}/{runnable-type}/{runnable-id}". In order to support schedules/ endpoint we might want to create separate handlers such as  "/apps/{app-id}/workflows/{workflow-id}", "/apps/{app-id}/flows/{flow-id}", "/apps/{app-id}/mapreduce/{mapreduce-id}" which would internally call the same function. This will avoid the collision with the schedule/ endpoint. Since it involves more changes I created jira (https://issues.cask.co/browse/CDAP-1155) for this to tackle this issue in the another PR. Please let me know if thats ok.